### DISCUSSION TITLE
Help about OsiClpSolverInterface.hpp

### DISCUSSION THREAD

**Discussion by xwendel** [coin-or/Cgl#78] [Q&A] [answered]
Hello everybody

I apologize if this is a dummy question. I am new about CGL. Fist, I have cloned and compiled Clp. After, I have cloned and compiled OSI. I did not find a way to link OSI and Clp. running

./configure --help

in the OSI directory, I saw OSI configure script does not have option to link Clp.

So, after I compiled OSI, I have cloned CGL. I have configured it using the command:

./configure ADD_CFLAGS="-O3 -march=native" ADD_CXXFLAGS="-O3 -march=native" --prefix=pwd--with-coinutils-lflags="-L/opt/coinutils/lib -lCoinUtils" --with-coinutils-cflags="-I/opt/coinutils/include/coin-or" --with-asl-lflags="-L/opt/asl/lib -lasl2" --with-asl-cflags="-I/opt/asl/include" --with-glpk-cflags="-I/opt/glpk/include" --with-glpk-lflags="-L/opt/glpk/lib -lglpk" --with-clp-cflags="-I/opt/clp/include" --with-clp-lflags="-L/opt/clp/lib -lClp" --with-osi-cflags="-I/opt/osi/include/coin-or" --with-osi-lflags="-L/opt/osi/lib -lOsi"

After run the configure script with success. I tried to compile and got the error:

make[1]: Entering directory '/opt/cgl_versions/cgl/build_gcc/src/CglClique'
CXX CglClique.lo
../../../src/CglClique/CglClique.cpp:865:10: fatal error: OsiClpSolverInterface.hpp: No such file or directory
865 | #include "OsiClpSolverInterface.hpp"
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

The file OsiClpSolverInterface.hpp is missing. I took a look in OSI and I could not find the file OsiClpSolverInterface.hpp . I suppose I am forgetting some CGL's dependency that includes this file. Could someone point me how I can fix my problem and compile CGL? Where can I find the file OsiClpSolverInterface.hpp?

Thanks in advanced

Wendel

**Comment by xwendel** [comment, marked as answer]
Oh, I think I found my mistake. I will describe here for the case someone has the same problem some day. Actually two mistakes

1 - It is necessary pass an OSI reference to CLP configure script to enable OSICLP.  
2 - In the CGL configure script I should have pass --with-clp-cflags="-I/opt/clp/include/coin-or" instead of --with-clp-cflags="-I/opt/clp/include"

Now, I have compiled CGL with success!

Thank you, guys!

Wendel

**Reply by tkralphs** [reply]
Is there a reason that you don't want to use [`coinbrew`](https://github.com/coin-or/coinbrew), as documented in the README? That is a far easier way to build. Just one command and everything is built and linked automagically.
